app/models/profile.rb

```diff
@@ -84,31 +83,37 @@ def in_account(account)
                     benchmark_id: benchmark_id)
   end
 
-  def add_rules(ids: nil, ref_ids: nil)
-    ProfileRule.import!(rule_ids_to_add(ids, ref_ids).map do |rule|
-      ProfileRule.new(profile_id: id, rule_id: rule.id)
-    end, ignore: true)
-  end
-
   def major_os_version
     benchmark ? benchmark.inferred_os_major_version : 'N/A'
   end
   alias os_major_version major_os_version
 
+  def update_rules(ids: nil, ref_ids: nil)
+    ids_to_add = rule_ids_to_add(ids, ref_ids)
+
+    ProfileRule.where(rule_id: rule_ids - ids_to_add, profile_id: id)
+               .destroy_all
+
+    ProfileRule.import!(ids_to_add.map do |rule|
+      ProfileRule.new(profile_id: id, rule_id: rule.id)
+    end)
+  end
```

Comment:
Renamed from `add_rules`. This method now intelligently updates the rules on the profile and will remove rules that aren't specified. Also, `ignore: true` was removed from the import, as we don't expect duplicates or any other errors anymore.

Reply:
This is actually pretty useful, I am working on making the frontend Edit Policy button bring up the wizard to allow re-tailoring and this comes in handy